### PR TITLE
Add implementation for critical-section 1.0

### PR DESCRIPTION
**Pull request by Dirbaio**
Picking up #433 since it seems stalled. Changes from #433 are:
- Update to `critical-section 1.0.0-alpha.2`
- Use `bool` restore token
- Name Cargo feature `critical-section-single-core`.

TODO before merging:

- [x] Wait for `critical-section 1.0` release https://github.com/rust-embedded/critical-section/pull/19

Co-Authored-By: Markus Reiter @reitermarkus

### CODE REVIEW

File: .github/workflows/clippy.yml
Comment (adamgreig) [inline .github/workflows/clippy.yml:4]: What's the intention with changing this and removing the `pull_request_target` entries below? Won't it stop the clippy annotations working since the secret key becomes unavailable?

File: src/critical_section.rs
Comment (adamgreig) [inline src/critical_section.rs:14]: Do we need to condition disabling interrupts on them currently being active, or can we unconditionally disable them (while still returning whether they _were_ active for `release()`)?

File: .github/workflows/clippy.yml
Comment (Dirbaio) [reply .github/workflows/clippy.yml]: oops, probably a rebase mistake. Fixed.

File: src/critical_section.rs
Comment (Dirbaio) [reply src/critical_section.rs]: it's what the PR did, can't think of a reason either. Changed it to do it unconditionally.

**Comment by reitermarkus** [comment]
> Picking up #433 since it seems stalled.

It's not stalled. I was just waiting for another review, same as my other open PRs. Feel free to ping me next time if you want me to rebase a PR.

Thanks anyways for picking this up.

The workflow `pull_request` change was meant to be temporary so CI can run on my fork.

File: src/interrupt.rs
Comment (adamgreig) [inline src/interrupt.rs:55]: Perhaps this method should now refer to critical-section, something like this:

```suggestion
/// Execute closure `f` in an interrupt-free context.
///
/// This method does not synchronise multiple cores and may disable required
/// interrupts on some platforms; see the critical-section crate for a cross-platform
/// way to enter a critical section which provides a `CriticalSection` token.
///
/// This crate provides an implementation for critical-section that simply disables
/// all interrupts which may be enabled with the `critical-section-single-core` feature.
```

Comment (Dirbaio) [reply src/interrupt.rs]: Done!

**Review by adamgreig — APPROVED** [review]
bors r+

**Comment by adamgreig** [comment]
bors retry

should go after #450 merges...

**Pull request merged**
